yellowbrick/features/base.py

```diff
@@ -104,21 +106,17 @@ def fit(self, X, y=None, **fit_params):
         This method must return self.
         """
 
-        # Get the shape of the data
-        nrows, ncols = X.shape
-
         # Handle the feature names if they're None.
         if self.features_ is None:
 
             # If X is a data frame, get the columns off it.
             if is_dataframe(X):
-                self.features_ = X.columns
+                self.features_ = np.array(X.columns)
```

Comment:
do we want to put support in here for structured / named arrays here at some point?

Reply:
Yes, we probably do; man, are they a pain in the butt though. I pulled out all of the code we had that used `X.view` since we were getting massive numbers of deprecation warnings from Numpy and I replaced it with `np.array(X.tolist())` which isn't great. 

StackOverflow post on the subject here if you have an answer: https://stackoverflow.com/questions/49085469/convert-structured-array-to-numpy-array-for-use-with-scikit-learn

Reply:
Added issue #319 in response

Reply:
well aware of how painful they can be to work with =)